src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.Utilities.fs

```diff
@@ -179,19 +180,20 @@ module internal Utilities =
 
             use p = new Process()
             p.StartInfo <- psi
-            p.Start() |> ignore
 
-            let stdOut = drainStreamToMemory p.StandardOutput
-            let stdErr = drainStreamToMemory p.StandardError
+            p.OutputDataReceived.Add(fun a -> outputDataReceived a.Data)
+            p.ErrorDataReceived.Add(fun a ->  errorDataReceived a.Data)
+
+            if p.Start() then
+                p.BeginOutputReadLine()
+                p.BeginErrorReadLine()
+                p.WaitForExit()
```

Comment:
It would also be useful to plumb a timeout parameter in here, to be passed in by the caller.